pkg/translator/sumoicmarshaler/sumo_ic_marshaler.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2022, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sumoicmarshaler
+
+import (
+	"encoding/json"
+	"errors"
+
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+type SumoICMarshaler struct {
+	format string
+}
+
+func (marshaler *SumoICMarshaler) Format() string {
+	return marshaler.format
+}
+
+func NewSumoICMarshaler() SumoICMarshaler {
+	return SumoICMarshaler{}
+}
+
+const (
+	SourceCategoryKey = "_sourceCategory"
+	SourceHostKey     = "_sourceHost"
+	SourceNameKey     = "log.file.path_resolved"
+)
+
+type field struct {
+}
+
+type icRecord struct {
+	Date           string `json:"data"`
+	SourceName     string `json:"sourceName"`
+	SourceHost     string `json:"sourceHost"`
+	SourceCategory string `json:"sourceCategory"`
+	Fields         field  `json:"fields"`
+	Message        string `json:"message"`
+}
+
+func transformLogsToIc(ld plog.Logs) ([]icRecord, error) {
+	icRecords := []icRecord{}
+	rls := ld.ResourceLogs()
+	for i := 0; i < rls.Len(); i++ {
+		rl := rls.At(i)
+		sourceCategory, exists := rl.Resource().Attributes().Get(SourceCategoryKey)
+		if exists == false {
+			return nil, errors.New("_sourceCategory attribute does not exists")
+		}
+		sourceHost, exists := rl.Resource().Attributes().Get(SourceHostKey)
+		if exists == false {
+			return nil, errors.New("_sourceHost attribute does not exists")
+		}
+		ills := rl.ScopeLogs()
+		for j := 0; j < ills.Len(); j++ {
+			ils := ills.At(j)
+			logs := ils.LogRecords()
+			for k := 0; k < logs.Len(); k++ {
+				lr := logs.At(k)
+				sourceName, exists := lr.Attributes().Get(SourceNameKey)
+				if exists == false {
+					return nil, errors.New("_sourceName attribute does not exists")
+				}
+				icRecords = append(icRecords, icRecord{lr.ObservedTimestamp().String(),
+					sourceName.AsString(), sourceHost.AsString(),
+					sourceCategory.AsString(), field{}, lr.Body().AsString()})
```

Comment:
Do we know what the format of `fields` should be? I'm pretty sure I know the values - it should be `rl.Attributes()`.